AnkiDroid/src/main/java/com/ichi2/anki/ImageOcclusionActivity.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright (c) 2024 Ashish Yadav <mailtoashish693@gmail.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify it under
+ *  the terms of the GNU General Public License as published by the Free Software
+ *  Foundation; either version 3 of the License, or (at your option) any later
+ *  version.
+ *
+ *  This program is distributed in the hope that it will be useful, but WITHOUT ANY
+ *  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ *  PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with
+ *  this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.ichi2.anki
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import kotlin.reflect.KClass
+import kotlin.reflect.jvm.jvmName
+
+/**
+ * ImageOcclusionActivity extends SingleFragmentActivity and serves as the activity for
+ * handling the Image Occlusion editor functionality and the behavior related to screen rotation
+ * in the Image Occlusion editor, where added, moved, resized, or deleted boxes do not properly
+ * adjust to the new aspect ratio.
```

Comment:
I can't undestand what you mean here.

Reply:
Doc or the file in general ?

Reply:
doc

Reply:
It means that this activity is for Image occlusion so that it handles the orientation changes when the editor (IO) has moved or resized (mask)

Reply:
1. adding `configChanges=""` without a `onConfigurationChanged` method means that the configuration changes are being **_ignored_**, not handled.
2. the actual issue is that the activity is recreated, which loses the webview state and happens with any activity with a webview. Not something specific to IO masks
3. You don't need to repeat the name of a class in its doc

```suggestion
* Handles adding and editing Image Occlusion cards.
* 
* Based in [SingleFragmentActivity], but with `configChanges="orientation|screenSize"`
* to avoid unwanted activity recreations
```

Reply:
`onConfigurationChanged` overrides nothing? I removed that due to this